### ISSUE TITLE
Script output is not printed to console 

### ISSUE THREAD

**Issue by brainelectronics** [mesonbuild/meson#8498]
**Describe the bug**
The stdout content is captured to a file if `capture:true` is set in a custom_target. Otherwise it won't be outputed anymore.

**To Reproduce**
Upgrading meson from 0.56.2 to 0.57.0 or newer introduced this bug/feature? with the commit 0626465ea8aa65b10776d5c4064e881fe0d6fa25 of fixing #8187 

The behaviour seems not to happen if I do not depend on other meson objects like e.g. an executable() which compiles, in my case 45 C objects beforehand.

```
python3 = find_program('python3', required : false)
my_custom_script = custom_target('Target_custom',
    command : [python3 , '@INPUT@',],
    # capture : true, # required since 0.57.0 as output is not printed to stdout, capture to file
    output : 'custom.log',
    input : 'random.py',
    depends: [binary_executable],
    build_by_default    : false
)
alias_target('Target_custom', my_custom_script)
```

random.py file looks similar to this.
```#!/usr/bin/env python
import logging
import sys

logging.basicConfig(level=logging.DEBUG, stream=sys.stdout)
logging.debug('This will get logged')
```

Output is:
`[45/45] Generating Target_custom with a custom command (wrapped by meson to set PATH)`

**Expected behavior**
`[45/45] Generating TargetA_custom with a custom command (wrapped by meson to set PATH)
DEBUG:root:This will get logged`

**system parameters**
* Is this a cross build
* Windows 10
* Python 3.8.5
* Meson 0.57.1
* Ninja 1.10.0.git.kitware.jobserver-1

PR will be created right after this ticket

**Comment by eli-schwartz** [comment]
> PR will be created right after this ticket

meson does not have a rule that people are required to open up an issue in addition to each PR. Feel free, next time, to just open the PR and include this description in the PR comments instead. :)

**Comment by kjellahl** [comment]
This is an issue not only for custom_target() but also for add_install_script().
stdout from an added install script disappears with Meson newer than 0.56.2.

**Comment by bruchar1** [comment]
Fixed by #11368.
